Automattic/wp-calypso
# My Plan: Add thank you header

Add "Thank You" UI to the `/plans/my-plan/SITE_SLUG` page for Jetpack sites when `?thank-you` URL param is present.

This is mostly just UI and some necessary querying and state selection. It will be associated with functionality that will arrive in another PR (#26690 or an alternative). This includes a static progress indicator. It is only for the "in progress" view of plan setup. The completed view is not included in this change.

This was broken out from #26690 to focus on the thank you banner.

See p6TEKc-2e8-p2

Closes #27039 (included)

## Screens

### iPhone 6+
![thank-you__iphone6plus](https://user-images.githubusercontent.com/841763/45098593-96fefe00-b125-11e8-8fbb-d54536d13a15.png)

### iPad Portrait
![thank-you__ipad-v](https://user-images.githubusercontent.com/841763/45098677-c0b82500-b125-11e8-94cb-d71446257fe6.png)

### iPad Landscape
![thank-you__ipad-h](https://user-images.githubusercontent.com/841763/45098590-96666780-b125-11e8-9738-866d537c6a71.png)

### Desktop
![thank-you](https://user-images.githubusercontent.com/841763/45098591-96fefe00-b125-11e8-9ba5-40cf1bef62eb.png)


## Testing
1. Visit https://calypso.live/plans/my-plan?thank-you&branch=add/jetpack-my-plan-header for a Jetpack site with a paid plan.
1. You'll see the thank you triggered when `?thank-you` is present on the URL.
1. Make sure the card looks good at different viewport sizes.
1. Post-checkout, you'll be sent to `plans/my-plan/SLUG?thank-you` and the card will be displayed  for Jetpack sites. It will not be displayed post-checkout for:
   - Jetpack site with no plan (no plan === no checkout)
   - Atomic site
   - Simple site
1. Try visiting the path directly for different sites and the rendered display should make sense.
